lib/remi/data_subjects/sftp_file.rb

```diff
@@ -1,4 +1,44 @@
 module Remi
+  module DataSubject::SftpFile
+
+    attr_reader :sftp_session
+
+    def sftp_retry(&block)
+      tries ||= 3
+
+      block.call
+    rescue RuntimeError => err
+      if (tries -= 1) > 0
+        logger.error "Error: #{err.message}"
+        logger.error "Will retry #{tries} more times"
+        sleep(1)
+        retry
+      else
+        raise err
+      end
+    end
+
+    def begin_connection
+      sftp_retry do
+        Timeout.timeout(30) do
```

Comment:
do you think this should be a passed in parameter? so it can default to 30 but if we have a particularly laggy day we could customize the amount of time before timeout?